docs/src/user_docs/assembly/cryptographic_operations.md

```diff
@@ -9,5 +9,5 @@ Miden assembly provides a set of instructions for performing common cryptographi
 | rpperm  <br> - *(1 cycle)*      | [C, B, A, ...]  | [F, E, D, ...] | $\{D, E, F\} \leftarrow permute(A, B, C)$ <br> where, $permute()$ computes a Rescue Prime Optimized permutation. |
 | rphash  <br> - *(16 cycles)*        | [B, A, ...]     | [C, ...]       | $C \leftarrow hash(A,B)$ <br> where, $hash()$ computes a 2-to-1 Rescue Prime Optimized hash. |
 | mtree_get  <br> - *(9 cycles)*     | [d, i, R, ...]  | [V, R, ...] |  Verifies that a Merkle tree with root $R$ opens to node $V$ at depth $d$ and index $i$. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. |
-| mtree_set <br> - *(14 cycles)*      | [d, i, R, V, ...] | [R', V, ...] | Updates a node in the Merkle tree with root $R$ at depth $d$ and index $i$ to value $V$. $R'$ is the Merkle root of the resulting tree. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. At the end of the operation Merkle tree in the advice provider with root $R$ is replaced with the Merkle tree with root $R'$. |
+| mtree_set <br> - *(14 cycles)*      | [d, i, R, V, ...] | [R', V, V', ...] | Updates a node in the Merkle tree with root $R$ at depth $d$ and index $i$ to value $V$. $R'$ is the Merkle root of the resulting tree and $V'$ is old value of the tree. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. At the end of the operation Merkle tree in the advice provider with root $R$ is replaced with the Merkle tree with root $R'$. |
```

Comment:
As a convention, we use the `'` to indicate new, so I think this will be a bit inconsistent with our existing docs. I think there are 2 ways to do this with more consistency, and I don't know which you prefer:

```
[d, i, R, V', ...] | [R', V', V, ...]
```

or 


```
[d, i, R, V, ...] | [R', V, V_old, ...]
```